pkg/controller/jobframework/reconciler.go

```diff
@@ -675,14 +676,27 @@ func (r *JobReconciler) getPodSetsInfoFromAdmission(ctx context.Context, w *kueu
 				return nil, err
 			}
 			for k, v := range flv.Spec.NodeLabels {
-				nodeSelector.NodeSelector[k] = v
+				podSetInfo.NodeSelectorOverwrite[k] = v
 			}
 			processedFlvs.Insert(flvName)
 		}
-
-		nodeSelectors[i] = nodeSelector
+		for _, admissionCheck := range w.Status.AdmissionChecks {
+			for _, podSetUpdate := range admissionCheck.PodSetUpdates {
+				if podSetUpdate.Name == podSetInfo.Name {
+					if err := podSetInfo.Merge(PodSetInfo{
+						Labels:       podSetUpdate.Labels,
+						Annotations:  podSetUpdate.Annotations,
+						Tolerations:  podSetUpdate.Tolerations,
+						NodeSelector: podSetUpdate.NodeSelector,
+					}); err != nil {
+						return nil, err
```

Comment:
`fmt.Errorf("in admission check %s: %w", admissionCheck.Name, err)`

Reply:
Done